controls/views.py

```diff
@@ -836,80 +836,65 @@ def import_component(request):
     result = ComponentImporter().import_components_as_json(import_name, oscal_component_json, request)
     return component_library(request)
 
+
+def raise_404_if_not_permitted_to_statement(request, statement, system_permission='view_system'):
+    """Raises a 404 if the user doesn't have the statement system permission"""
+    while True:
+        if request.user.is_superuser or request.user.is_staff:
+            # Allow access if the user has superuser or staff member permissions respectively
+            break
+        else:
+            # Allow access if the user has the statement system permission
+            try:
+                system = System.objects.get(root_element=statement.consumer_element)
+                if request.user.has_perm(system_permission, system):
+                    break
+            except System.DoesNotExist:
+                pass
+
+        raise Http404
+
+def raise_404_if_not_permitted_to_change_statement(request, statement):
+    """Raises a 404 if the user doesn't have permission to change the statement"""
+    raise_404_if_not_permitted_to_statement(request, statement, 'change_system')
```

Comment:
Since this is a one-line function I think it would make this functionality cleaner to just use `raise_404_if_not_permitted_to_statement` directly instead of nesting it in this function.